java/jakarta/el/ELContext.java

```diff
@@ -139,6 +140,10 @@ public void setLocale(Locale locale) {
      * @since EL 3.0
      */
     public void addEvaluationListener(EvaluationListener listener) {
+        if (listeners == null) {
```

Comment:
Is it possible that this method is called by multiple threads ? In that case you will need to add locking here.

Reply:
i dont think so. At least in JSF and JSP it should be used "request scoped". Also currently it doesnt use concurrent list impl.